File: corecomply-mvp/api/CoreComply.Api/Domain/Entities/Risk.cs
namespace CoreComply.Api.Domain.Entities;

public class Risk
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty; // operational, financial, compliance, strategic
    public string Likelihood { get; set; } = "medium"; // low, medium, high
    public string Impact { get; set; } = "medium"; // low, medium, high, critical
    public string Status { get; set; } = "open"; // open, mitigating, closed
    public string Owner { get; set; } = string.Empty;
    public DateTime IdentifiedDate { get; set; } = DateTime.UtcNow;
    public DateTime? ReviewDate { get; set; }
    public string? Mitigation { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: corecomply-mvp/web/src/types/risk.ts
// Risk entity types matching backend CoreComply.Api.Domain.Entities.Risk

export interface Risk {
  id: number;
  title: string;
  description: string;
  category: string; // operational, financial, compliance, strategic
  likelihood: string; // low, medium, high
  impact: string; // low, medium, high, critical
  status: string; // open, mitigating, closed
  owner: string;
  identifiedDate: string;
  reviewDate?: string | null;
  mitigation?: string | null;
  createdAt: string;
}

export interface CreateRiskRequest {
  title: string;
  description: string;
  category: string;
  likelihood: string;
  impact: string;
  owner: string;
  reviewDate?: string | null;
}

export interface UpdateRiskRequest {
  title: string;
  description: string;
  category: string;
  likelihood: string;
  impact: string;
  status: string;
  owner: string;
  reviewDate?: string | null;
  mitigation?: string | null;
}

// Helper function to calculate risk rating from likelihood and impact
export function calculateRiskRating(likelihood: string, impact: string): string {
  const likelihoodScore: Record<string, number> = {
    low: 1,
    medium: 2,
    high: 3
  };

  const impactScore: Record<string, number> = {
    low: 1,
    medium: 2,
    high: 3,
    critical: 4
  };

  const lScore = likelihoodScore[likelihood.toLowerCase()] || 2;
  const iScore = impactScore[impact.toLowerCase()] || 2;
  const total = lScore * iScore;

  if (total >= 9) return 'Critical';
  if (total >= 6) return 'High';
  if (total >= 4) return 'Medium';
  return 'Low';
}

// Helper function to get risk rating color for badges
export function getRiskRatingColor(rating: string): "default" | "destructive" | "secondary" | "outline" {
  switch (rating.toLowerCase()) {
    case 'critical':
      return 'destructive';
    case 'high':
      return 'destructive';
    case 'medium':
      return 'secondary';
    case 'low':
      return 'outline';
    default:
      return 'secondary';
  }
}

// Helper function to get status color for badges
export function getStatusColor(status: string): "default" | "destructive" | "secondary" | "outline" {
  switch (status.toLowerCase()) {
    case 'open':
      return 'destructive';
    case 'mitigating':
      return 'secondary';
    case 'closed':
      return 'outline';
    default:
      return 'secondary';
  }
}
